src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/BaseCodeActionService.cs

```diff
@@ -76,10 +80,12 @@ protected async Task<IEnumerable<AvailableCodeAction>> GetAvailableCodeActions(I
             await CollectCodeFixesActions(document, span, codeActions);
             await CollectRefactoringActions(document, span, codeActions);
 
+            var distinctActions = codeActions.GroupBy(x => x.Title).Select(x => x.First());
+
             // Be sure to filter out any code actions that inherit from CodeActionWithOptions.
             // This isn't a great solution and might need changing later, but every Roslyn code action
             // derived from this type tries to display a dialog. For now, this is a reasonable solution.
-            var availableActions = ConvertToAvailableCodeAction(codeActions)
+            var availableActions = ConvertToAvailableCodeAction(distinctActions)
```

Comment:
We need to investigate in (the future) the ability to trigger some of the diaglog elements through native inputs granted by something like vscode / atom / etc.

Reply:
I checked this little bit about month ago. If i remembered correctly points which seems abstractions against VS specific stuffs in "extract interface" for example like IDialogProvider / SomeAbstractProvider (i don't remember exact classes however) they are set as internal. So i don't think we can create instance of CodeActionProviders where are depencies to them (seems most of actions with dialogs / options). I could try write issues to roslyn repo about those later when i have time to find exact classes from Roslyn codebase.